infrastructure/repository/phone_book_repository_test.go

```diff
@@ -0,0 +1,144 @@
+package repositories_test
+
+import (
+	"fmt"
+	"testing"
+
+	repositories "github.com/cyneptic/letsgo-smspanel/infrastructure/repository"
+	"github.com/cyneptic/letsgo-smspanel/internal/core/entities"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+// Set up the test environment by creating a repository and a test phone book
+func setupTestEnvironment(t *testing.T) (*repositories.PGRepository, *entities.PhoneBook) {
+	repo := repositories.NewGormDatabase()
+
+	phoneBook := entities.PhoneBook{
+		Name:   "test phone book",
+		UserId: uuid.New(),
+	}
+
+	// Create a test phone book in the repository
+	_, err := repo.CreatePhoneBook(phoneBook)
+	assert.NoError(t, err)
+
+	// Retrieve the created phone book
+	var pBook entities.PhoneBook
+	err = repo.DB.Where("user_id = ?", phoneBook.UserId).First(&pBook).Error
+	assert.NoError(t, err)
+	phoneBook.ID = pBook.ID
+
+	return repo, &phoneBook
+}
+
+// Clean up the test environment by deleting the created phone book
+func cleanupTestEnvironment(t *testing.T, repo *repositories.PGRepository, phoneBook *entities.PhoneBook) {
+	err := repo.DB.Unscoped().Delete(phoneBook).Error
+	assert.NoError(t, err)
+}
+
+// Test creating a phone book
+func TestCreatePhoneBook(t *testing.T) {
+	repo, phoneBook := setupTestEnvironment(t)
+	defer cleanupTestEnvironment(t, repo, phoneBook)
+
+	// Query the repository and assert the resultsz
+	var pBook entities.PhoneBook
+	err := repo.DB.Where("user_id = ?", phoneBook.UserId).First(&pBook).Error
+	assert.NoError(t, err)
+	assert.Equal(t, phoneBook.UserId, pBook.UserId)
+	assert.Equal(t, phoneBook.Name, pBook.Name)
+}
+
+// Test getting a list of phone books
+func TestGetPhoneBookList(t *testing.T) {
+	repo, phoneBook := setupTestEnvironment(t)
+	defer cleanupTestEnvironment(t, repo, phoneBook)
+
+	// Get the list of phone books from the repository and assert the results
+	pBooks, err := repo.GetPhoneBookList(*phoneBook)
+	assert.Nil(t, err)
+	assert.NotNil(t, pBooks)
+	assert.Len(t, pBooks, 1)
+}
+
+// Test getting a list of phone books with an invalid user ID
+func TestGetPhoneBookListWithInvalidUserId(t *testing.T) {
+	repo, phoneBook := setupTestEnvironment(t)
+	defer cleanupTestEnvironment(t, repo, phoneBook)
+
+	// Define a phone book with an invalid user ID
+	invalidUserId := uuid.New()
+	var invalidPhoneBook entities.PhoneBook
+	invalidPhoneBook.UserId = invalidUserId
+
+	// Get the list of phone books with the invalid phone book and assert the results
+	pBooks, err := repo.GetPhoneBookList(invalidPhoneBook)
+
+	fmt.Println(pBooks)
+	assert.Nil(t, err)
```

Comment:
if something invalid id passed to a function it should error. we need error information to know what has went wrong.